### ISSUE TITLE
Best practice to import attachments with jiraone export file

### ISSUE THREAD

**Issue by kalupator** [princenyeche/jiraone#144]
Hello!
In the prepared by jiraone csv\json export file, links to attachments from the old jira look like this:
https://myjira.domain.com/secure/attachment/11853/MyFile.docx

But when importing a file into a new Jira, the attachments file cannot be downloaded from the old Jira, because authentication is required in the old Jira.

I have already dealt with this before and solved this problem in the following way:
For each attachment link I added a login and password for old jira in the following way: ?os_username=USER&os_password=PWD.

For example:
https://myjira.domain.com/secure/attachment/11853/MyFile.docx?os_username=USER&os_password=PWD

After that, when importing a file in the new Jira, attachments were downloaded from the old Jira.

I want to understand if there is any solution for jiraone when preparing the export file?
What is the best practice for importing attachments with jiraone prepared export file?
Thanks!

**Comment by princenyeche** [comment]
Hey @kalupator 

To do that in `jiraone` for a CSV import, you have to export the Jira issues using the below argument. For example
```python
# previous import statement
issue_export(jql=jql, extension="csv", final_file="export-test", allow_media=True)
```
The argument `allow_media` with the value `True` adds your credential to every media file within the CSV export of the attachment columns, so you can easily import attachment files with your exported file. I hope that helps.

**Comment by kalupator** [comment]
Thanks for the answer! I will check it as soon as possible and write the result.

**Comment by kalupator** [comment]
@princenyeche, I tried to install allow_media=True.
Now all links contain a password:
`https://e:PASSWORD@oldjira.domain.com/secure/attachment/11698/file.docx`
But I get an error during import:
`Error:` HTTP status code: 401 Unauthorized
I think it's because the link only has a password but no user.

**Comment by princenyeche** [comment]
@kalupator I believe that's the username. If your site allows usernames, that should be how to auth it to allow attachments extracted from one site to another.

**Comment by kalupator** [comment]
@princenyeche 
After several experiments, I realized that it is the first letter of the username.
If i replace it with the full username, authorization does indeed occur correctly.
But for some reason jiraone always export only the first letter of the username in attachments links.
Do you know what the problem might be?

**Comment by princenyeche** [comment]
Yeah, I found the problem [here](https://github.com/princenyeche/jiraone/blob/main/src/jiraone/reporting.py#L4141). The code uses an index when it should not. I've fixed it and will push the fix in the next update.

**Comment by kalupator** [comment]
@princenyeche 
Great news!
And when do you plan to do this? I'm planning a big migration of my projects after the holidays =)

**Comment by princenyeche** [comment]
I'll push it together with #122 issue since it's still there. The PR is already raised #149 just waiting to confirm the other issue is resolved then can push it together.
